### DISCUSSION TITLE
Where is the Makefile?

### DISCUSSION THREAD

**Discussion by GPD1** [jamulussoftware/jamulus#2219] [Technical Support & Help] [answered]
Hi,
Trying to compile for updating a headless server (Ubuntu), but there is no Makefile in the latest distribution, so the compiling instructions don't work...  (There used to be one in version 3.7)
Thx!

**Comment by rdica** [comment, marked as answer]
The short of it:
`qmake "CONFIG+=noupcasename headless nosound" Jamulus.pro && make clean && make`

Per https://github.com/jamulussoftware/jamulus/blob/master/COMPILING.md link in https://jamulus.io/wiki/Installation-for-Linux

**Reply by GPD1** [reply]
Sorry, I was blindly following the instructions and it was failing on the first step ("make distclean"), as I had not run the second step ("qmake "CONFIG+=nosound headless"" ) yet...

**Reply by pljones** [reply]
:)...  `make distclean; qmake 'CONFIG+=headless nosound' && make -j3 && make clean` (or whatever CONFIG) -- that way the distclean can fail if there's nothing to do and you clean up at the end.

**Comment by ann0see** [comment]
Note, you don’t need to compile the server anymore. We provide deb files.

**Reply by GPD1** [reply]
I could not find deb files for nosound/headless Ubuntu servers (that only have the minimum qtcore, qtnetwork, qtconcurrent and qtxml packages), or is it that I didn't look right?

**Reply by rdica** [reply]
@ann0see Do you know if there's any interest in providing other Linux pkg formats?
I started compiling from source because I use Fedora servers and clients and there are no RPMs available from this project.
I also build on Ubuntu now as well so I can test betas and RCs.

@GPD1 https://github.com/jamulussoftware/jamulus/releases
https://github.com/jamulussoftware/jamulus/releases/download/r3_8_1/jamulus_headless_3.8.1_ubuntu_amd64.deb

**Reply by pljones** [reply]
@rdica  We're not _currently_ planning to extend packaged support - due to the fact we've not finalised the .deb support and want to get that done first.

We'd need to re-design the Linux builds if we were to build for multiple package formats -- the buld time runs too long as is, we couldn't just run another stream for RPM, we'd have to do it as part of the Linux build but produce multiple packaged results.  That might mean quite a lot of changes to how we currently do the Linux (and, in fact, because of how the builds work - all) builds.

Indeed, all the release builds are here:
https://github.com/jamulussoftware/jamulus/releases/tag/r3_8_1 at the bottom of the page
